### ISSUE TITLE
Track/adjust for test failure with genno 1.23.0

### ISSUE THREAD

**Issue by khaeru** [iiasa/message_ix#788]
(Apparently) with genno 1.23.0, we see failures like [this](https://github.com/iiasa/message_ix/actions/runs/7676514776/job/20924167977#step:9:420):
```
>       assert_qty_equal(obs, demand, check_attrs=False)

message_ix/tests/test_report.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

left = Index(['chicago', 'new-york', 'topeka'], dtype='object', name='n')
right = MultiIndex([( 'chicago',),
            ('new-york',),
            (  'topeka',)],
           names=['n'])
obj = 'Series.index'

    def _check_types(left, right, obj: str = "Index") -> None:
        if not exact:
            return
    
>       assert_class_equal(left, right, exact=exact, obj=obj)
E       AssertionError: Series.index are different
E       
E       Series.index classes are different
E       [left]:  Index(['chicago', 'new-york', 'topeka'], dtype='object', name='n')
E       [right]: MultiIndex([( 'chicago',),
E                   ('new-york',),
E                   (  'topeka',)],
E                  names=['n'])
```

This may actually be a regression in `genno` due to some housekeeping in khaeru/genno#115, that adapted AttrSeries.\_\_init__ to address deprecation warnings with pandas 2.2. What appears to have happened is that the code now skips [this block](https://github.com/khaeru/genno/blob/6cedebc53c46cd6df2f8c3bb2a665c96d8f1d4cf/genno/core/attrseries.py#L160-L168) that ensures a pd.MultiIndex. So `obs` in this test, while *functionally* identical to `demand`, has a different index class and thus fails assert_qty_equal().

This fix is probably upstream, this is just to track.

**Comment by khaeru** [comment]
@glatterf42 my preferred temporarily mitigation here would be to install `genno<1.23` in the CI workflows.

**Comment by glatterf42** [comment]
We currently have [two failed tests](https://github.com/iiasa/message_ix/actions/runs/7691428691/job/20956736573), but when all tests [first started failing](https://github.com/iiasa/message_ix/actions/runs/7676514776/job/20924167780), there was only one. This was on Saturday and I found these version changes:

- dask 2024.1.0 -> 2024.1.1
- genno 1.22.0 -> 1.23.0
- coverage 7.4.0 -> 7.4.1

On Sunday, the second test failure was introduced when some additional dependency updates came along:

- overrides 7.6.0 -> 7.7.0
- pytest 7.4.4 -> 8.0.0

**Comment by glatterf42** [comment]
The relevant changes in pytest might be detailed here: https://docs.pytest.org/en/stable/changelog.html#old-deprecations-are-now-errors

**Comment by glatterf42** [comment]
Partially/temporarily addressed by #789.

**Comment by khaeru** [comment]
This should be addressed in genno 1.23.1, just released.
